source/NVDAObjects/UIA/wordDocument.py

```diff
@@ -716,40 +716,65 @@ def event_UIA_notification(self, activityId=None, **kwargs):
-			ui.message(_("Navigating by sentence not supported in this document"))
-			gesture.send()
-			return
-		# Using the legacy object model,
-		# Move the caret to the next sentence in the requested direction.
-		legacyInfo = LegacyWordDocumentTextInfo(self, textInfos.POSITION_CARET)
-		legacyInfo.move(textInfos.UNIT_SENTENCE, direction)
-		# Save the start of the sentence for future use
-		legacyStart = legacyInfo.copy()
-		# With the legacy object model,
-		# Move the caret to the end of the new sentence.
-		legacyInfo.move(textInfos.UNIT_SENTENCE, 1)
-		legacyInfo.updateCaret()
-		# Fetch the caret position (end of the next sentence) with UI automation.
-		endInfo = self.makeTextInfo(textInfos.POSITION_CARET)
-		# Move the caret back to the start of the next sentence,
-		# where it should be left for the user.
-		legacyStart.updateCaret()
-		# Fetch the new caret position (start of the next sentence) with UI Automation.
-		startInfo = self.makeTextInfo(textInfos.POSITION_CARET)
-		# Make a UI automation text range spanning the entire next sentence.
-		info = startInfo.copy()
-		info.end = endInfo.end
+
+		info = None
+
+		# Prefer UIA remote sentence navigation when available.
+		if UIARemote.isSupported():
+			try:
+				caretInfo = self.makeTextInfo(textInfos.POSITION_CARET)
+				sentenceRange = UIARemote.msWord_moveTextRangeBySentence(
+					self.UIAElement,
+					caretInfo._rangeObj,
+					direction,
+				)
+			except Exception:
+				log.debugWarning(
+					"Failed to fetch caret text range for remote sentence navigation",
+					exc_info=True,
+				)
+			else:
+				if sentenceRange is not None:
+					info = WordDocumentTextInfo(self, textInfos.POSITION_CARET, _rangeObj=sentenceRange)
+					info.updateCaret()
+
+		if info is None:
+			if not self.WinwordSelectionObject:
+				# Legacy object model not available.
+				# Translators: a message when navigating by sentence is unavailable in MS Word
+				ui.message(_("Navigating by sentence not supported in this document"))
+				gesture.send()
+				return
+			# Using the legacy object model,
+			# Move the caret to the next sentence in the requested direction.
+			legacyInfo = LegacyWordDocumentTextInfo(self, textInfos.POSITION_CARET)
+			legacyInfo.move(textInfos.UNIT_SENTENCE, direction)
+			# Save the start of the sentence for future use
+			legacyStart = legacyInfo.copy()
+			legacyInfo.move(textInfos.UNIT_SENTENCE, 1)
+			# Fetch the caret position (end of the next sentence) with UI automation.
+			endInfo = self.makeTextInfo(textInfos.POSITION_CARET)
+			# Move the caret back to the start of the next sentence,
+			# where it should be left for the user.
+			legacyStart.updateCaret()
+			# Fetch the new caret position (start of the next sentence) with UI Automation.
+			startInfo = self.makeTextInfo(textInfos.POSITION_CARET)
+			# Make a UI automation text range spanning the entire next sentence.
+			info = startInfo.copy()
+			info.end = endInfo.end
```

Comment:
can we move this to a helper function? e.g. `getLegacyTextInfo(self, direction: int) -> LegacyWordDocumentTextInfo`